### PR TITLE
Signature verification for in-toto via verify-blob

### PR DESCRIPTION
**Pull request by zregvart**
For in-toto the data being signed is in the DSSE protocol format[1], so
when verifying using `cosign verify-blob` we need to present the data in
that format.

[1] https://github.com/secure-systems-lab/dsse/blob/master/protocol.md

### CODE REVIEW

**Review by lcarva — APPROVED** [review]
Verified on my local setup. LGTM

File: hack/chains/check-taskrun.sh [hunk 1]
```diff
@@ -114,29 +120,17 @@ say $SIGNATURE
...
 
 # For debugging...
 title "To view taskrun"
 say " env EDITOR=view kubectl edit $TASKRUN_NAME"
 
 # Clean up
 rm $SIG_FILE $PAYLOAD_FILE
-
-# Use the exit code from cosign
-exit $COSIGN_EXIT_CODE
```
Comment (simonbaird) [inline hack/chains/check-taskrun.sh:142]: The reason for the set +e and exit code hoop jumping was to ensure the `rm $SIG_FILE ..` clean up happened if the verify failed.

Probably not that important, but I remember now why I did that...

**Comment by simonbaird** [comment]
One nitpick: Could mention the Jira in the commit, it's https://issues.redhat.com/browse/HACBS-89 .

**Review by sbose78 — APPROVED** [review]

File: hack/chains/check-taskrun.sh (hunk 1 above)
Comment (lcarva) [reply hack/chains/check-taskrun.sh:142]: Ah ok. In that case, maybe `trap` right after the files are created is a good idea:
```bash
# Cosign needs files on disk to do a verify-blob afaict
SIG_FILE=$( mktemp )
PAYLOAD_FILE=$( mktemp )
trap "rm -f ${SIG_FILE} ${PAYLOAD_FILE}" EXIT
echo -n "$PAYLOAD" > $PAYLOAD_FILE
echo -n "$SIGNATURE" > $SIG_FILE
```

Comment (simonbaird) [reply hack/chains/check-taskrun.sh:142]: Yeah that's sensible, but imo it doesn't particularly need to be in this PR.

Comment (zregvart) [reply hack/chains/check-taskrun.sh:142]: I've added a [trap](https://github.com/redhat-appstudio/infra-deployments/pull/183/files#diff-0aaca8058d25ea8b1a3abd31c7d3eb390726d1ab896ef5829ad0b67b5e187390R114-R117) to cleanup the temp files.

**Comment by zregvart** [comment]
I've modified the commit message to mention [HACBS-89](https://issues.redhat.com/browse/HACBS-89)

**Comment by simonbaird** [comment]
Looks good to me - nice patch!

**Pull request merged**